homeassistant/components/tessie/number.py

```diff
@@ -0,0 +1,155 @@
+"""Number platform for Tessie integration."""
+from __future__ import annotations
+
+from tessie_api import set_charge_limit, set_charging_amps, set_speed_limit
+
+from homeassistant.components.number import NumberDeviceClass, NumberEntity, NumberMode
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    PERCENTAGE,
+    PRECISION_WHOLE,
+    UnitOfElectricCurrent,
+    UnitOfSpeed,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import TessieDataUpdateCoordinator
+from .entity import TessieEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Tessie sensor platform from a config entry."""
+    coordinators = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        EntityClass(coordinator)
+        for EntityClass in (
+            TessieChargeLimitSocNumberEntity,
+            TessieSpeedLimitModeNumberEntity,
+            TessieCurrentChargeNumberEntity,
+        )
+        for coordinator in coordinators
+    )
+
+
+class TessieCurrentChargeNumberEntity(TessieEntity, NumberEntity):
```

Comment:
I feel like entity descriptions would add value here, wdyt?

Reply:
The reason I didnt is because async_set_native_value is more complex for each type, and the min max values are sometimes static and sometimes dynamic. The description would have ended up being rather complex, so I kept them as classes.

Reply:
The reason I didnt is because async_set_native_value is more complex for each entity, and the min max values are sometimes static and sometimes dynamic. The description would have ended up being rather complex, so I kept them as classes.

Reply:
Ok, challenge accepted, I have used entitiy descriptions. Wasnt that hard after all. 
```
await self.run(
            self.entity_description.func, **{self.entity_description.argument: value}
        )
        ```